config/yaml_test.go

```diff
@@ -351,3 +351,27 @@ func TestMerge(t *testing.T) {
 		})
 	}
 }
+
+func TestNilYAMLProviderSetDefaultTagValue(t *testing.T) {
+	type Inner struct {
+		Set bool `yaml:"set" default:"true"`
+	}
+	data := struct {
+		ID0 int             `yaml:"id0" default:"10"`
+		ID1 string          `yaml:"id1" default:"string"`
+		ID2 Inner           `yaml:"id2"`
+		ID3 []Inner         `yaml:"id3"`
+		ID4 map[Inner]Inner `yaml:"id4"`
+		ID5 *Inner          `yaml:"id5"`
```

Comment:
sure. and I already know that will break before I even add it... Thanks for pointing that out.
Will add `[6]Inner` and `[7]*Inner`